requirements/RequirementsChecker.py

```diff
@@ -176,32 +179,124 @@ def do_polygons_intersect(self, polygon_a, polygon_b):
+			
+			theta_max = np.degrees(theta_max)
+			collision_zone = None
+			relative_angle = round(np.degrees(relative_angle),2)
+
+			#assign a collision zone
+			if (-theta_max <= relative_angle <= theta_max):
+				collision_zone = "front"
+				return collision_zone, relative_angle
+			elif 180 - theta_max <= relative_angle or relative_angle <= -180 + theta_max:
+				collision_zone = "rear"
+				return collision_zone, relative_angle
+			else:
+				#side points 
+				side_length = np.sqrt((x_arc_front[0] - x_arc_back[-1]) ** 2 + (y_arc_front[0] - y_arc_back[-1]) ** 2)
+				num_of_samples = math.ceil(side_length / (2*ped_rad))
+
+				x_right = np.linspace(x_arc_front[0], x_arc_back[-1], num_of_samples)
+				y_right = np.linspace(y_arc_front[0], y_arc_back[-1], num_of_samples)
+
+				x_left = np.linspace(x_arc_front[-1], x_arc_back[0], num_of_samples)
+				y_left = np.linspace(y_arc_front[-1], y_arc_back[0], num_of_samples)
+
+				#do not include points from the arc
+				x_right = x_right[1:-1]
+				y_right = y_right[1:-1]
+				x_left = x_left[1:-1]
+				y_left = y_left[1:-1]
+
+				#left and right
+				all_x_coord = np.concatenate((x_right, x_left))
+				all_y_coord = np.concatenate((y_right, y_left))
+
+				distances = np.sqrt((all_x_coord - ped_x) ** 2 + (all_y_coord - ped_y) ** 2)
+				min_dist = np.min(distances)
+				min_dist_index = np.argmin(distances)
+
+				x_collided = all_x_coord[min_dist_index]
+				y_collided = all_y_coord[min_dist_index]
+			
+				if min_dist <= ped_rad:
+					#find the vector of the point of collision with the center of the vehicle
+					collision_vector = np.array([x_collided - center_x, y_collided - center_y])
+
+					#finding angle between vector between collision vector and vehicle direction
+					vehicle_direction = np.array([np.cos(yaw), np.sin(yaw)])
+					dot_product = np.dot(vehicle_direction, collision_vector)
+					denom = np.sqrt(((collision_vector[0])**2 + (collision_vector[1])**2))*np.sqrt((vehicle_direction[0])**2 + (vehicle_direction[1])**2)
+
+					relative_angle = np.arccos(dot_product/denom)
+					cross_product = vehicle_direction[0] * collision_vector[1] - vehicle_direction[1] * collision_vector[0]
+
+					if cross_product < 0:
+						relative_angle = -relative_angle
+					
+					theta_max = np.degrees(theta_max)
+					relative_angle = round(np.degrees(relative_angle), 2)
 
-		#generates 10 points given the angle of the arc
-		theta_arc = np.linspace(theta_1, theta_2, 10)
-		x_arc = center_x + arc_radius * np.cos(theta_arc)
-		y_arc = center_y + arc_radius * np.sin(theta_arc)
+					#assign a collision zone
+					if 0 > relative_angle:
+						collision_zone = "right"
+					else:
+						print("collision")
+						collision_zone = "left"
 
-		#check if the arc collides with pedestrian
-		distances = np.sqrt((x_arc - ped_x) ** 2 + (y_arc - ped_y) ** 2)
-		min_dist = np.min(distances)
+					return collision_zone, relative_angle
 	
-		return min_dist <= ped_rad
+		return False, None
```

Comment:
on 297, you return zone, angle but here False, None. Shouldn't it be None, None?